pkg/manifest/jsonmanifest/entry.go

```diff
@@ -5,71 +5,42 @@
 package jsonmanifest
 
 import (
-	"encoding/json"
 	"net/http"
 
 	"github.com/ethersphere/bee/pkg/manifest"
 	"github.com/ethersphere/bee/pkg/swarm"
 )
 
-// verify JSONEntry implements manifest.Entry.
-var _ manifest.Entry = (*JSONEntry)(nil)
+// verify jsonEntry implements manifest.Entry.
+var _ manifest.Entry = (*jsonEntry)(nil)
 
-// JSONEntry is a JSON representation of a single manifest entry for a JSONManifest.
-type JSONEntry struct {
-	reference swarm.Address
-	name      string
-	headers   http.Header
+// jsonEntry is a JSON representation of a single manifest entry for a jsonManifest.
+type jsonEntry struct {
+	R swarm.Address `json:"reference"` // reference
+	N string        `json:"name"`      // name
+	H http.Header   `json:"headers"`   // header
 }
 
-// NewEntry creates a new JSONEntry struct and returns it.
-func NewEntry(reference swarm.Address, name string, headers http.Header) *JSONEntry {
-	return &JSONEntry{
-		reference: reference,
-		name:      name,
-		headers:   headers,
+// NewEntry creates a new jsonEntry struct and returns it.
+func NewEntry(reference swarm.Address, name string, headers http.Header) manifest.Entry {
+	return &jsonEntry{
+		R: reference,
+		N: name,
+		H: headers,
 	}
 }
 
 // Reference returns the address of the file in the entry.
-func (me *JSONEntry) Reference() swarm.Address {
-	return me.reference
+func (me *jsonEntry) Reference() swarm.Address {
+	return me.R
 }
 
 // Name returns the name of the file in the entry.
-func (me *JSONEntry) Name() string {
-	return me.name
+func (me *jsonEntry) Name() string {
+	return me.N
 }
 
 // Headers returns the headers for the file in the manifest entry.
```

Comment:
comment says `headers` but this returns just one `header` no?

Reply:
correct. i've changed _all_ references to singular (that was fun...!)